### PR TITLE
Update Crashlytics Report Proto with new Rollouts fields

### PR DESCRIPTION
**Pull request by themiswang**
- Update CrashlyticsReport with new proto, followed the backend change: [cl/554467862](http://cl/554467862)
- Adding RolloutAssignmentList to UserMetadata
- Adding limits to parameter value for rollout assignment

#no-changelog

### CODE REVIEW

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignment.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 @Encodable
 @AutoValue
 public abstract class RolloutAssignment {
+  private static final int MAX_PARAMETER_VALUE = 256;
```
Comment (danasilver) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignment.java:30]: Maybe add a `_LENGTH` or `_SIZE` suffix?

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignment.java [hunk 2]
```diff
@@ -58,7 +65,13 @@ static RolloutAssignment create(String json) throws JSONException {
     String parameterValue = dataObj.getString("parameterValue");
     String variantId = dataObj.getString("variantId");
     long templateVersion = dataObj.getLong("templateVersion");
+
+    String validatedParameterValue = parameterValue;
+    if (validatedParameterValue.length() > MAX_PARAMETER_VALUE_LENGTH) {
+      validatedParameterValue = validatedParameterValue.substring(0, MAX_PARAMETER_VALUE_LENGTH);
+    }
```
Comment (samedson) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignment.java:72]: Being a little nitpicky - could we put the validation in a method? So we can say `String validatedParameterValue = validate(parameterValie)` both here and up at line 51?

**Review by samedson — APPROVED** [review]
Just one suggestion

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignment.java (hunk 2 above)
Comment (themiswang) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignment.java:76]: Done!

**Pull request merged**